packages/mobile/src/components/audio/Audio.tsx

```diff
@@ -36,49 +41,83 @@ import { logListen } from './listens'
 
 const { getUser } = cacheUsersSelectors
 const { getPlaying, getSeek, getCurrentTrack, getCounter } = playerSelectors
-const { getIndex, getLength, getRepeat, getShuffle, getShuffleIndex } =
-  queueSelectors
+const { getRepeat } = queueSelectors
 const { getIsReachable } = reachabilitySelectors
 
 const { getUserId } = accountSelectors
 
+type ProgressData = {
+  currentTime: number
+  duration?: number
+}
+
+// TODO: Probably don't use global for this
 declare global {
   // eslint-disable-next-line no-var
-  var progress: {
-    currentTime: number
-    playableDuration?: number
-    seekableDuration?: number
-  }
+  var progress: ProgressData
 }
 
 const SKIP_DURATION_SEC = 15
-
 const RECORD_LISTEN_SECONDS = 1
 
-const styles = StyleSheet.create({
-  backgroundVideo: {
-    position: 'absolute',
-    display: 'none',
-    top: 0,
-    left: 0,
-    bottom: 0,
-    right: 0
-  }
-})
+// Perform initial setup for the track player
+const setupTrackPlayer = async () => {
+  await TrackPlayer.setupPlayer()
+
+  // Set options for controlling music on the lock screen when the app is in the background
+  TrackPlayer.updateOptions({
+    // Media controls capabilities
+    capabilities: [
+      Capability.Play,
+      Capability.Pause,
+      Capability.SkipToNext,
+      Capability.SkipToPrevious,
+      Capability.Stop,
+      Capability.SeekTo
+    ],
+    // Capabilities that will show up when the notification is in the compact form on Android
+    compactCapabilities: [
+      Capability.Play,
+      Capability.Pause,
+      Capability.SkipToNext,
+      Capability.SkipToPrevious
+    ],
+    // Notification form capabilities
+    notificationCapabilities: [
+      Capability.Play,
+      Capability.Pause,
+      Capability.SkipToNext,
+      Capability.SkipToPrevious
+    ],
+    android: {
+      // Default behavior for Android
+      appKilledPlaybackBehavior: AppKilledPlaybackBehavior.ContinuePlayback
+    }
+  })
+}
+
+const playerEvents = [
+  Event.PlaybackError,
+  Event.PlaybackQueueEnded,
+  Event.PlaybackProgressUpdated,
+  Event.RemotePlay,
+  Event.RemotePause,
+  Event.RemoteNext,
+  Event.RemotePrevious,
+  Event.RemoteSeek
+]
 
 export const Audio = () => {
   const { isEnabled: isStreamMp3Enabled } = useFeatureFlag(
     FeatureFlags.STREAM_MP3
   )
+  const progress = useProgress(100) // 100ms update interval
```

Comment:
Love this!! Will this cause a rerender of audio component every 100ms though?

Reply:
chatted with @sliptype , 100ms was the polling interval that we used before so i just copied that over. Can increase it to 250ms if there are issues.
This would only affect the purple progress bar of the closed now playing drawer so not a big deal to increase this for sure

Reply:
Gotcha makes sense, main thing is cost of rerendering audio component that much? Dunno maybe we check profiler at some point